.github/actions/restore-node/check-git-status.sh

```diff
@@ -1,11 +1,26 @@
 #!/bin/bash
+
 # Set verbose execution
 set -x
+
 # Navigate to the specified directory
 cd "$1" || exit $?
+
+# Get the value of IGNORE_ENDO_BRANCH
+ignore_dirty_yarn_lock=$2 || exit $?
+
 # Check for unexpected changes
 # Fail if git status detects changes
 changes=$(git status --porcelain)
+
+if [ "$ignore_dirty_yarn_lock" = true ]; then
+  # Remove yarn.lock from the changes list
+  # This exclusion is necessary because when a PR is merged into the master branch,
+  # the 'yarn.lock' file is changed when Endo integration test CI workflow is executed.
+  # For more details, refer to https://github.com/Agoric/agoric-sdk/issues/9850.
+  changes=$(echo "$changes" | grep -v " yarn.lock")
+fi
```

Comment:
```suggestion
if [ "$ignore_dirty_yarn_lock" = true ]; then
  # When integration is requested with a specific Endo branch, ignore changes:
  # - package.json has a modified `"resolutions":` field,
  # - yarn.lock is changed because of those resolutions.
  # For more details, refer to https://github.com/Agoric/agoric-sdk/issues/9850.
  changes=$(echo "$changes" | grep -vE '^.. (package\.json|yarn\.lock)$')
fi
```